package/MDAnalysis/lib/_cutil.pyx

```diff
@@ -49,6 +49,119 @@ ctypedef cset[int] intset
 ctypedef cmap[int, intset] intmap
 
 
+@cython.boundscheck(False) # turn off bounds-checking for entire function 
+@cython.wraparound(False)  # turn off negative index wrapping for entire function 
+cdef void _inverse_unique_unsorted_contiguous_array_inplace(
+						np.intp_t[::1] full_arr,
+                				np.intp_t[::1] unique_arr,
+                                		np.intp_t[::1] mask
+						):
+    """Find the inverse of a unique 1D array to map onto a non-unique 1D array.
+
+    Cython driver function. Loops though the unique array and finds the indices 
+    in the full array that correspond to the values in the unique array.
+
+    ie: np.array([1,2,3])[np.array(0,0,1,2,2)] == np.array([1,1,2,3,3])
```

Comment:
Could you reformat this as an example, i.e. using the Example section? This currently doesn't really tell me anything, as I don't know what `full_arr` and `unique_arr[mask]` are meant to correspond to (on the line below).

```
Example
--------
::
    full_arr = np.array([1, 1, 2, 3, 3])
    ...
```


Typically `Examples` will be under the `Parameters` and `Returns` sections of a docstring

Reply:
This would still be nice.